pkg/module/go_get_fetcher.go

```diff
@@ -0,0 +1,163 @@
+	if repoURI == "" {
+		return nil, errors.New("invalid repository identifier")
+	}
+
+	return &goGetFetcher{
+		fs:           fs,
+		repoURI:      repoURI,
+		version:      version,
+		goBinaryName: goBinaryName,
+	}, nil
+}
+
+// Fetch downloads the sources and returns path where it can be found. This function will
+// always return a ref, even if it returns a non-nil error.
+//
+// If an error was returned, the returned ref's Read method will return an error, but you
+// should always call ref.Clear() on the returned ref
+func (g *goGetFetcher) Fetch(mod, ver string) (Ref, error) {
+	var ref Ref
+	ref = noopRef{}
+
+	// setup the GOPATH
+	goPathRoot, err := afero.TempDir(g.fs, "", "athens")
+	if err != nil {
+		// TODO: return a ref for cleaning up the goPathRoot
+		// https://github.com/gomods/athens/issues/329
+		return ref, err
+	}
+	sourcePath := filepath.Join(goPathRoot, "src")
+	modPath := filepath.Join(sourcePath, getRepoDirName(g.repoURI, g.version))
+	if err := g.fs.MkdirAll(modPath, os.ModeDir|os.ModePerm); err != nil {
+		// TODO: return a ref for cleaning up the goPathRoot
+		// https://github.com/gomods/athens/issues/329
+		return ref, err
+	}
+
+	// setup the module with barebones stuff
+	if err := prepareStructure(g.fs, modPath); err != nil {
+		// TODO: return a ref for cleaning up the goPathRoot
+		// https://github.com/gomods/athens/issues/329
+		return ref, err
+	}
+
+	cachePath, err := getSources(g.goBinaryName, g.fs, goPathRoot, modPath, mod, ver)
+	if err != nil {
+		// TODO: return a ref that cleans up the goPathRoot
+		// https://github.com/gomods/athens/issues/329
+		return newDiskRef(g.fs, cachePath, ver), err
+	}
+	// TODO: make sure this ref also cleans up the goPathRoot
+	// https://github.com/gomods/athens/issues/329
+	ref = newDiskRef(g.fs, cachePath, ver)
+
+	return ref, err
+}
+
+// Hacky thing makes vgo not to complain
+func prepareStructure(fs afero.Fs, repoRoot string) error {
+	// vgo expects go.mod file present with module statement or .go file with import comment
+	gomodPath := filepath.Join(repoRoot, "go.mod")
+	gomodContent := []byte("module mod")
+	if err := afero.WriteFile(fs, gomodPath, gomodContent, 0666); err != nil {
+		return err
+	}
+
+	sourcePath := filepath.Join(repoRoot, "mod.go")
+	sourceContent := []byte(`package mod // import "mod"`)
+	return afero.WriteFile(fs, sourcePath, sourceContent, 0666)
+}
+
+// given a filesystem, gopath, repository root, module and version, runs 'vgo get'
+// on module@version from the repoRoot with GOPATH=gopath, and returns the location
+// of the module cache. returns a non-nil error if anything went wrong. always returns
+// the location of the module cache so you can delete it if necessary
+func getSources(goBinaryName string, fs afero.Fs, gopath, repoRoot, module, version string) (string, error) {
+	version = strings.TrimPrefix(version, "@")
+	if !strings.HasPrefix(version, "v") {
```

Comment:
from my previous review: 

you shouldn't need this whole block in the first place. version is *always* going to include v and therefore you can remove the whole `if !strings.HasPrefix(version, "v")` block. If you really don't trust the caller, then calling `semver.IsValid` is probably more wholesome. But imo, just going out and saying `go get module@version` is good enough because Go will validate the version anyway

Reply:
(instead of relying on `go get` to tell us that the version is invalid when we call)